EIPS/blockhash_refactoring.md

```diff
@@ -0,0 +1,84 @@
+### Preamble
+
+    EIP: <to be assigned>
+    Title: Blockhash refactoring
+    Author: Vitalik Buterin
+    Type: Standard Track
+    Category: Core
+    Status: Draft
+    Created: 2017-02-10
+    
+### Summary
+
+Stores blockhashes in the state, reducing the protocol complexity and the need for client implementation complexity in order to process the BLOCKHASH opcode. Also extends the range of how far back blockhash checking can go, with the side effect of creating direct links between blocks with very distant block numbers, facilitating much more efficient initial light client syncing.
+
+### Parameters
+
+* `METROPOLIS_FORK_BLKNUM`: TBD
+* `SUPER_USER`: 2**160 - 2
+* `BLOCKHASH_CONTRACT_ADDR`: 0xf0 (ie. 240)
+* `BLOCKHASH_CONTRACT_CODE`: see below
+
+### Specification
+
+If `block.number == METROPOLIS_FORK_BLKNUM`, then when processing the block, before processing any transactions set the code of BLOCKHASH_CONTRACT_ADDR to BLOCKHASH_CONTRACT_CODE.
+
+If `block.number >= METROPOLIS_FORK_BLKNUM`, then when processing a block, before processing any transactions execute a call with the parameters:
```

Comment:
Should this call increment the nonce?  I feel ambiguous https://github.com/ethereum/cpp-ethereum/pull/4066#discussion_r117254932